beacon-chain/p2p/gossipsub_peer_crawler.go

```diff
@@ -0,0 +1,550 @@
+package p2p
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/OffchainLabs/prysm/v7/beacon-chain/p2p/gossipsubcrawler"
+	"github.com/pkg/errors"
+	"golang.org/x/sync/semaphore"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+
+	"github.com/ethereum/go-ethereum/p2p/enode"
+)
+
+type peerNode struct {
+	isPinged bool
+	node     *enode.Node
+	peerID   peer.ID
+	topics   map[string]struct{}
+}
+
+type crawledPeers struct {
+	mu              sync.RWMutex
+	peerNodeByEnode map[enode.ID]*peerNode
+	peerNodeByPid   map[peer.ID]*peerNode
+	pidsByTopic     map[string]map[peer.ID]struct{}
+}
+
+func (cp *crawledPeers) updateStatusToPinged(enodeID enode.ID) {
+	cp.mu.Lock()
+	defer cp.mu.Unlock()
+
+	existingPNode, ok := cp.peerNodeByEnode[enodeID]
+	if !ok {
+		return
+	}
+
+	// we only want to ping a node with a given NodeId once -> not on every sequence number change
+	// as ping is simply a test of a node being reachable and not fake
+	existingPNode.isPinged = true
+}
+
+func (cp *crawledPeers) updateCrawledIfNewer(node *enode.Node, topics []string) (shouldPing bool, err error) {
+	if node == nil {
+		return false, errors.New("node is nil")
+	}
+
+	return cp.updatePeer(node, topics)
+}
+
+func (cp *crawledPeers) updatePeer(node *enode.Node, topics []string) (shouldPing bool, err error) {
```

Comment:
You define named returns, but don't use them.

```suggestion
func (cp *crawledPeers) updatePeer(node *enode.Node, topics []string) (bool, error) {
```